cmd/bnsapi/bnsclient.go

```diff
@@ -0,0 +1,206 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	weaveapp "github.com/iov-one/weave/app"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/orm"
+)
+
+// BnsClient is implemented by any service that provides access to BNS API.
+type BnsClient interface {
```

Comment:
Just suggestion: You can rename `BnsClient` to `WeaveAPIClient` or something similar to provide a base library for future Weave users.

Reply:
There are at least 3 weave clients already. I think that's enough :wink: